compliance_checker/base.py

```diff
@@ -191,9 +191,10 @@ def __del__(self):
         are cleared before the next checker uses it. Some caches were
         inadvertently mutated by other functions.
         """
-
-        cfutil.get_geophysical_variables.cache_clear()
-        cfutil.get_time_variables.cache_clear()
+        # odd errors -- module getting deleted before this object?
+        if cfutil is not None:
+            cfutil.get_geophysical_variables.cache_clear()
+            cfutil.get_time_variables.cache_clear()
```

Comment:
I'd like to keep these changes in another PR so we can investigate the cause and, if this is the best solution. Let's keep the context here to just remove the last bits of pkg_resources.

Reply:
Sure -- this is an odd one. I wonder if the __del__ is still needed at all.

Reply:
@benjwadams if you have some cycles can give us some guidance here?

Reply:
This is really weird -- `cfutil` is a module that was imported at the top of the module. And yet it's set to None at least once when the tests are run - maybe there's a mock or something in the tests?

Sorry -- no time to dig deeper right now.

-CHB